zebedee-cms/src/main/java/com/github/onsdigital/zebedee/dataset/api/DatasetAPIClient.java

```diff
@@ -0,0 +1,87 @@
+package com.github.onsdigital.zebedee.dataset.api;
+
+import com.github.onsdigital.zebedee.configuration.Configuration;
+import com.github.onsdigital.zebedee.content.util.ContentUtil;
+import com.github.onsdigital.zebedee.verification.http.ClientConfiguration;
+import com.github.onsdigital.zebedee.verification.http.PooledHttpClient;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+
+import static com.github.onsdigital.zebedee.logging.ZebedeeLogBuilder.logInfo;
+
+/**
+ * HTTP DatasetAPIClient for the dataset API.
+ */
+public class DatasetAPIClient implements DatasetClient {
+
+    private static DatasetClient instance;
+    private static String datasetAPIURL = Configuration.getDatasetAPIURL();
+    private static PooledHttpClient client;
+
+    private DatasetAPIClient() {
+        // private constructor - use getInstance()
+    }
+
+    /**
+     * Get the singleton instance of the DatasetAPIClient.
+     *
+     * @return
+     */
+    public static DatasetClient getInstance() {
+
+        if (DatasetAPIClient.instance == null) {
+            synchronized (DatasetAPIClient.class) {
+                if (DatasetAPIClient.instance == null) {
+                    DatasetAPIClient.instance = new DatasetAPIClient();
+                    DatasetAPIClient.client = new PooledHttpClient(DatasetAPIClient.datasetAPIURL, new ClientConfiguration());
+                }
+            }
+        }
+        return DatasetAPIClient.instance;
+    }
+
+    /**
+     * Get the dataset for the given dataset ID.
+     *
+     * @param datasetID
+     * @return
+     */
+    @Override
+    public Dataset getDataset(String datasetID) throws IOException {
+
+        String path = "/datasets/" + datasetID;
+
+        logInfo("Calling dataset API").addParameter("path", path).log();
+
+        try (CloseableHttpResponse response = client.sendGet(path, null, null)) {
```

Comment:
I think you want to check the http status in case its an error

Reply:
The client does this internally and throws exceptions

Reply:
The client doesn't check for a specific response code it only checks its not > 302 there is no way to say what response status was expected or specially what the status codes mean

Reply:
Updated to check for specific response codes.